### PR TITLE
raidboss: r1s left/right, tb, and aoe triggers

### PR DESCRIPTION
**Pull request by Legends0**
Adds a left/right call for boss. There are multiple spell-ids, could use a double-check to make sure the ones here are both from the boss and not the other entities.

TB callout just calls the initial target.

AoE callout for bloody scratch

### CODE REVIEW

**Comment by valarnin** [comment]
> Adds a left/right call for boss. There are multiple spell-ids, could use a double-check to make sure the ones here are both from the boss and not the other entities.

My timeline PR should be able to shed some light on what each ID corresponds to. tl;dr: cast vs damage 1 vs damage 2, and then some alternate IDs for the clones.

---

I have a bunch of triggers for this as well. No overlap with what you've already written, but can't PR them without a conflict and haven't had a chance to actually test them in-instance yet.

Here's the triggerset file, if you want to test them and include them in this PR. Otherwise I'll do them in a different PR when I get around to testing them (probably testing Thursday night, then PR will be Friday sometime).

```ts
import Outputs from '../../../../../resources/outputs';
import { Responses } from '../../../../../resources/responses';
import { Directions } from '../../../../../resources/util';
import ZoneId from '../../../../../resources/zone_id';
import { RaidbossData } from '../../../../../types/data';
import { NetMatches } from '../../../../../types/net_matches';
import { TriggerSet } from '../../../../../types/trigger';

export interface Data extends RaidbossData {
  lastPawprintTarget?: string;
  actorSetPosTracker: { [id: string]: NetMatches['ActorSetPos'] };
  mouserMatchedTile?: (typeof mapEffectData)[keyof typeof mapEffectData]['location'];
}

// MapEffect tile map:
// 00 01 02 03
// 04 05 06 07
// 08 09 0A 0B
// 0C 0D 0E 0F

const mapEffectTileState = {
  'cracked': '00020001',
  'clear': '00040004',
  'broken': '00200010',
  'refreshing': '00800004', // refreshing from cracked
  'rebuilding': '01000004', // rebuilding from broken
} as const;

const mapEffectData = {
  '00': {
    'location': '00',
    'centerX': 85,
    'centerY': 85,
    ...mapEffectTileState,
  },
  '01': {
    'location': '01',
    'centerX': 95,
    'centerY': 85,
    ...mapEffectTileState,
  },
  '02': {
    'location': '02',
    'centerX': 105,
    'centerY': 85,
    ...mapEffectTileState,
  },
  '03': {
    'location': '03',
    'centerX': 115,
    'centerY': 85,
    ...mapEffectTileState,
  },
  '04': {
    'location': '04',
    'centerX': 85,
    'centerY': 95,
    ...mapEffectTileState,
  },
  '05': {
    'location': '05',
    'centerX': 95,
    'centerY': 95,
    ...mapEffectTileState,
  },
  '06': {
    'location': '06',
    'centerX': 105,
    'centerY': 95,
    ...mapEffectTileState,
  },
  '07': {
    'location': '07',
    'centerX': 115,
    'centerY': 95,
    ...mapEffectTileState,
  },
  '08': {
    'location': '08',
    'centerX': 85,
    'centerY': 105,
    ...mapEffectTileState,
  },
  '09': {
    'location': '09',
    'centerX': 95,
    'centerY': 105,
    ...mapEffectTileState,
  },
  '0A': {
    'location': '0A',
    'centerX': 105,
    'centerY': 105,
    ...mapEffectTileState,
  },
  '0B': {
    'location': '0B',
    'centerX': 115,
    'centerY': 105,
    ...mapEffectTileState,
  },
  '0C': {
    'location': '0C',
    'centerX': 85,
    'centerY': 115,
    ...mapEffectTileState,
  },
  '0D': {
    'location': '0D',
    'centerX': 95,
    'centerY': 115,
    ...mapEffectTileState,
  },
  '0E': {
    'location': '0E',
    'centerX': 105,
    'centerY': 115,
    ...mapEffectTileState,
  },
  '0F': {
    'location': '0F',
    'centerX': 115,
    'centerY': 115,
    ...mapEffectTileState,
  },
} as const;

const headMarkerData = {
  // Vfx Path: tank_lockon02k1
  tankbuster: '00DA',
  // Vfx Path: lockon8_t0w
  lineStack: '00F4',
  // Vfx Path: loc05sp_05a_se_p
  spreadMarker: '0178',
  // Vfx Path: m0884_vanish_7sec_p1
  pawprint: '021A',
} as const;

const triggerSet: TriggerSet<Data> = {
  id: 'AacLightHeavyweightM1Savage',
  zoneId: ZoneId.AacLightHeavyweightM1Savage,
  timelineFile: 'r1s.txt',
  initData: () => ({
    actorSetPosTracker: {},
  }),
  triggers: [
    {
      id: 'R1S ActorSetPos Collector',
      type: 'ActorSetPos',
      netRegex: { id: '4[0-9A-F]{7}', capture: true },
      run: (data, matches) => {
        data.actorSetPosTracker[matches.id] = matches;
      },
    },
    {
      id: 'R1S Mouser',
      type: 'StartsUsing',
      netRegex: { id: '996C', capture: true },
      condition: (data, matches) => {
        const actorSetPosLine = data.actorSetPosTracker[matches.sourceId];
        if (actorSetPosLine === undefined)
          return false;
        const x = parseFloat(actorSetPosLine.x);
        const y = parseFloat(actorSetPosLine.y);

        const loc = Object.values(mapEffectData)
          .find((tile) =>
            Math.abs(tile.centerX - x) < 1 && Math.abs(tile.centerY - y) < 1
          );
        if (loc === undefined)
          return false;

        const tile = loc.location;

        if (tile !== '09' && tile !== '0A')
          return false;

        data.mouserMatchedTile = tile;
        return true;
      },
      // We don't need a suppressSeconds since only one of the SW/SE tiles will get hit twice
      durationSeconds: 11,
      infoText: (data, _matches, output) => {
        const dangerTile = data.mouserMatchedTile;
        if (dangerTile === undefined)
          return false;

        // Danger tile is SW, so safe movement is SW => SE (Stay)
        if (dangerTile === '09') {
          return output.swSeStay!({
            dir1: output['dirSW']!(),
            sep: output.separator!(),
            dir2: output['dirSE']!(),
          });
        }

        const dirs = ['dirSW', 'dirSE', 'dirSW'].map((e) => output[e]!());

        return output.combo!({ dirs: dirs.join(output.separator!()) });
      },
      run: (data) => delete data.mouserMatchedTile,
      outputStrings: {
        ...Directions.outputStrings8Dir,
        swSeStay: {
          en: '${dir1} ${sep} ${dir2} (Stay)',
        },
        separator: {
          en: ' => ',
          de: ' => ',
          ja: ' => ',
          cn: ' => ',
        },
        combo: {
          en: '${dirs}',
          de: '${dirs}',
          ja: '${dirs}',
          cn: '${dirs}',
        },
      },
    },
    {
      id: 'R1S Headmarker Line Stack',
      type: 'HeadMarker',
      netRegex: { id: headMarkerData.lineStack, capture: true },
      suppressSeconds: 5,
      response: Responses.stackMarkerOn(),
    },
    {
      id: 'R1S Headmarker Spread Markers',
      type: 'HeadMarker',
      netRegex: { id: headMarkerData.spreadMarker, capture: true },
      suppressSeconds: 5,
      response: Responses.spread(),
    },
    {
      id: 'R1S Headmarker Pawprint Collector',
      type: 'HeadMarker',
      netRegex: { id: headMarkerData.pawprint, capture: true },
      run: (data, matches) => data.lastPawprintTarget = matches.target,
    },
    {
      id: 'R1S Elevate and Eviscerate Launch Self',
      type: 'StartsUsing',
      netRegex: { id: '9446', source: 'Copy Cat', capture: false },
      condition: (data) => data.me === data.lastPawprintTarget,
      infoText: (_data, _matches, output) => output.text!(),
      outputStrings: {
        text: {
          en: 'Launch Forward (Aim for uncracked tile)',
        },
      },
    },
    {
      id: 'R1S Elevate and Eviscerate Launch Other',
      type: 'StartsUsing',
      netRegex: { id: '9446', source: 'Copy Cat', capture: false },
      condition: (data) => data.me !== data.lastPawprintTarget,
      infoText: (data, _matches, output) => output.text!({ target: data.lastPawprintTarget }),
      outputStrings: {
        text: {
          en: '${target} Launch',
        },
      },
    },
    {
      id: 'R1S Elevate and Eviscerate Stun Self',
      type: 'StartsUsing',
      netRegex: { id: '9448', source: 'Copy Cat', capture: false },
      condition: (data) => data.me === data.lastPawprintTarget,
      infoText: (_data, _matches, output) => output.text!(),
      outputStrings: {
        text: {
          en: 'Stand on uncracked tile',
        },
      },
    },
    {
      id: 'R1S Elevate and Eviscerate Stun Other',
      type: 'StartsUsing',
      netRegex: { id: '9448', source: 'Copy Cat', capture: false },
      condition: (data) => data.me !== data.lastPawprintTarget,
      infoText: (data, _matches, output) => output.text!({ target: data.lastPawprintTarget }),
      outputStrings: {
        text: {
          en: '${target} Stun',
        },
      },
    },
    {
      id: 'R1S Quadruple Swipe',
      type: 'StartsUsing',
      netRegex: { id: '945D', source: 'Black Cat', capture: false },
      infoText: (_data, _matches, output) => output.partner!(),
      outputStrings: {
        partner: {
          en: 'Partner Stacks',
          de: 'Partner sammeln',
          fr: 'Package partenaire',
          ja: 'ペア',
          cn: '和搭档分摊',
          ko: '파트너 쉐어',
        },
      },
    },
    {
      id: 'R1S Delayed Quadruple Swipe',
      type: 'StartsUsing',
      netRegex: { id: '945D', source: 'Black Cat', capture: false },
      delaySeconds: 24.4,
      infoText: (_data, _matches, output) => output.partner!(),
      outputStrings: {
        partner: {
          en: 'Partner Stacks',
          de: 'Partner sammeln',
          fr: 'Package partenaire',
          ja: 'ペア',
          cn: '和搭档分摊',
          ko: '파트너 쉐어',
        },
      },
    },
    {
      id: 'R1S Double Swipe',
      type: 'StartsUsing',
      netRegex: { id: '945F', source: 'Black Cat', capture: false },
      infoText: (_data, _matches, output) => output.healerStacks!(),
      outputStrings: {
        healerStacks: Outputs.healerGroups,
      },
    },
    {
      id: 'R1S Delayed Double Swipe',
      type: 'StartsUsing',
      netRegex: { id: '945F', source: 'Black Cat', capture: false },
      delaySeconds: 24.4,
      infoText: (_data, _matches, output) => output.healerStacks!(),
      outputStrings: {
        healerStacks: Outputs.healerGroups,
      },
    },
    {
      id: 'R1S Overshadow',
      type: 'StartsUsing',
      netRegex: { id: '9497', source: 'Black Cat', capture: true },
      response: Responses.stackMarkerOn(),
    },
    {
      id: 'R1S Splintering Nails',
      type: 'StartsUsing',
      netRegex: { id: '9499', source: 'Black Cat', capture: false },
      infoText: (_data, _matches, output) => output.rolePositions!(),
      outputStrings: {
        rolePositions: {
          en: 'Role positions',
          de: 'Rollenposition',
          fr: 'Positions par rôle',
          ja: 'ロール特定位置へ',
          cn: '去指定位置',
          ko: '직업별 산개위치로',
        },
      },
    },
  ],
};

export default triggerSet;
```

File: ui/raidboss/data/07-dt/raid/r1s.ts [hunk 1]
```diff
@@ -8,7 +9,34 @@ const triggerSet: TriggerSet<Data> = {
...
+      id: 'R1S Biscuit Maker',
+      type: 'StartsUsing',
+      netRegex: { id: '9495', source: 'Black Cat', capture: true },
+      response: Responses.tankBuster(),
+    },
+    {
+      id: 'R1S Bloody Scratch',
+      type: 'StartsUsing',
+      netRegex: { id: '9494', source: 'Black Cat', capture: false },
+      response: Responses.aoe(),
```
Comment (xiashtra) [inline ui/raidboss/data/07-dt/raid/r1s.ts:37]:
```suggestion
      response: Responses.bigAoe(),
```
This is unsurvivable without mitigation in day 1 gear

**Review by xiashtra — APPROVED** [review]
These basic triggers look ok to merge as is; if you incorporate @valarnin's additions, there are linting errors that will need to be resolved and the additional triggers should be re-reviewed.

**Comment by xiashtra** [comment]
I think these should get PRed separately for clean-up/review.

**Comment by xiashtra** [comment]
> > Adds a left/right call for boss. There are multiple spell-ids, could use a double-check to make sure the ones here are both from the boss and not the other entities.
> 
> My timeline PR should be able to shed some light on what each ID corresponds to. tl;dr: cast vs damage 1 vs damage 2, and then some alternate IDs for the clones.
> 
> I have a bunch of triggers for this as well. No overlap with what you've already written, but can't PR them without a conflict and haven't had a chance to actually test them in-instance yet.
> ```

I think these should get PRed separately for clean-up/review.

File: ui/raidboss/data/07-dt/raid/r1s.ts (hunk 1 above)
Comment (Legends0) [reply ui/raidboss/data/07-dt/raid/r1s.ts]: Added, thanks for the suggestion.

**Pull request merged**